### PR TITLE
estimator_status minor cleanup

### PR DESCRIPTION
**Pull request by dagar**
Semi related to https://github.com/PX4/Firmware/issues/9757.

### CODE REVIEW

**Comment by TSC21** [comment]
@dagar can we hold thid while the msg restructure PR I am doing is not in? This can be a complement of it.

File: msg/estimator_status.msg [hunk 1]
```diff
@@ -1,14 +1,11 @@
 float32[24] states		# Internal filter states
-float32 n_states		# Number of states effectively used
+uint8 n_states		# Number of states effectively used
 
 float32[3] vibe			# IMU vibration metrics in the following array locations
 # 0 : Gyro delta angle coning metric = filtered length of (delta_angle x prev_delta_angle)
 # 1 : Gyro high frequency vibe = filtered length of (delta_angle - prev_delta_angle)
 # 2 : Accel high frequency vibe = filtered length of (delta_velocity - prev_delta_velocity)
 
-uint8 nan_flags			# Bitmask to indicate NaN states
```
Comment (bkueng) [inline msg/estimator_status.msg]: Note: this will need to be removed from flight revew too

**Comment by dagar** [comment]
@TSC21 I'd rather keep moving, but I'll volunteer to handle any rebasing that's nasty. There's a lot more I'd like to do with estimator_status, and it's overall going to be less painful to do it in small pieces incrementally.

**Comment by dagar** [comment]
Simplified and rebased on master.

File: src/modules/ekf2/ekf2_main.cpp [hunk 2]
```diff
@@ -1251,7 +1252,6 @@ void Ekf2::run()
 			_ekf.get_ekf_soln_status(&status.solution_status_flags);
 			_ekf.get_imu_vibe_metrics(status.vibe);
 			status.time_slip = _last_time_slip_us / 1e6f;
-			status.nan_flags = 0.0f; // unused
 			status.health_flags = 0.0f; // unused
```
Comment (TSC21) [inline src/modules/ekf2/ekf2_main.cpp]: Should be set to NaN, IMO

File: src/modules/mavlink/mavlink_messages.cpp [hunk 3]
```diff
@@ -2003,33 +2003,37 @@ class MavlinkStreamLocalPositionNEDCOV : public MavlinkStream
...
-			msg.vy = est.states[4];
-			msg.vz = est.states[5];
-			msg.ax = est.states[6];
-			msg.ay = est.states[7];
-			msg.az = est.states[8];
-
-			for (int i = 0; i < 9; i++) {
-				msg.covariance[i] = est.covariances[i];
-			}
+			// HACK for LPE, this is completely wrong for EKF2
```
Comment (TSC21) [inline src/modules/mavlink/mavlink_messages.cpp]: This is not needed anymore after https://github.com/PX4/Firmware/pull/9895

Comment (TSC21) [reply src/modules/mavlink/mavlink_messages.cpp]: https://github.com/PX4/Firmware/pull/9895/files#diff-478580be4572c29aaaa7e5861078259eR648

File: src/modules/ekf2/ekf2_main.cpp (hunk 2 above)
Comment (dagar) [reply src/modules/ekf2/ekf2_main.cpp]: These are uint8_t.

**Pull request merged**